src/cascadia/TerminalSettingsEditor/Actions.h

```diff
@@ -4,32 +4,85 @@
 #pragma once
 
 #include "Actions.g.h"
+#include "KeyBindingViewModel.g.h"
 #include "ActionsPageNavigationState.g.h"
+#include "RebindKeysEventArgs.g.h"
 #include "Utils.h"
+#include "ViewModelHelpers.h"
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
-    struct CommandComparator
+    struct KeyBindingViewModelComparator
     {
-        bool operator()(const Model::Command& lhs, const Model::Command& rhs) const
+        bool operator()(const Editor::KeyBindingViewModel& lhs, const Editor::KeyBindingViewModel& rhs) const
         {
             return lhs.Name() < rhs.Name();
         }
     };
 
+    struct RebindKeysEventArgs : RebindKeysEventArgsT<RebindKeysEventArgs>
+    {
+    public:
+        RebindKeysEventArgs(const Control::KeyChord& oldKeys, const Control::KeyChord& newKeys) :
+            _OldKeys{ oldKeys },
+            _NewKeys{ newKeys } {}
+
+        WINRT_PROPERTY(Control::KeyChord, OldKeys, nullptr);
+        WINRT_PROPERTY(Control::KeyChord, NewKeys, nullptr);
+    };
+
+    struct KeyBindingViewModel : KeyBindingViewModelT<KeyBindingViewModel>, ViewModelHelper<KeyBindingViewModel>
+    {
+    public:
+        KeyBindingViewModel(const Control::KeyChord& keys, const Settings::Model::Command& cmd);
+
+        hstring Name() const { return _Command.Name(); }
+        hstring KeyChordText() const { return _KeyChordText; }
+        Settings::Model::Command Command() const { return _Command; };
+
+        // UIA Text
+        hstring EditButtonName() const noexcept;
+        hstring CancelButtonName() const noexcept;
+        hstring AcceptButtonName() const noexcept;
+        hstring DeleteButtonName() const noexcept;
+
+        void EnterHoverMode() { IsHovered(true); };
+        void ExitHoverMode() { IsHovered(false); };
+        void FocusContainer() { IsContainerFocused(true); };
+        void UnfocusContainer() { IsContainerFocused(false); };
+        void FocusEditButton() { IsEditButtonFocused(true); };
+        void UnfocusEditButton() { IsEditButtonFocused(false); };
+        bool ShowEditButton() const noexcept;
+        void ToggleEditMode();
+        void DisableEditMode() { IsInEditMode(false); }
+        void AttemptAcceptChanges();
+        void AttemptAcceptChanges(hstring newKeyChordText);

```

Comment:
this is scary

Reply:
What does one do that the other doesn't?

Reply:
The non-parameter version is used in Actions.xaml. We can't just use the parameter version here because the `{x:Bind AttemptAcceptChanges(Text)}` trick
1. cannot access `_ProposedKeys`
2. might be able to smartly get the text box's text, but can't be used because it needs to return _something_

The parameter version is used here:
https://github.com/microsoft/terminal/blob/27582a9186c39ea9f5caf070e08fd80e50e1d587/src/cascadia/TerminalSettingsEditor/Actions.cpp#L144-L147

We need to directly extract the text from the text box, so we _need_ some kind of override version.